src/plugins/BunqLayoutPlugin/BunqLayoutPlugin.js

```diff
@@ -0,0 +1,173 @@
+// @flow
+import BunqLayout from "./components/BunqLayout/BunqLayout";
+import Sidebar from "./components/Sidebar/Sidebar";
+
+/**
+ * @type {Object}
+ */
+const actions = {
+    /**
+     * @param {Object} apiSpecification
+     *
+     * @returns {Object}
+     */
+    updateSidebarEndpoints: (apiSpecification: Object): Object => {
+        const tags = [];
+
+        if (apiSpecification.tags) {
+            for (const tag of apiSpecification.tags) {
+                tags.push(tag.name);
+            }
+        }
+
+        return {
+            type: "BUNQ_UPDATE_SIDEBAR_ENDPOINTS",
+            payload: tags
+        }
+    },
+
+    /**
+     * @param {Function} originalAction
+     * @param {Object} system
+     *
+     * @returns {Object}
+     */
+    updateApiSpecification: (
+        originalAction: Function,
+        system: Object
+    ): Object => (apiSpecification: Object): Object => {
+        system.bunqActions.updateSidebarEndpoints(JSON.parse(apiSpecification));
+
+        return originalAction(apiSpecification);
+    },
+
+    /**
+     * @param {Function} originalAction
+     * @param {Object} system
+     */
+    readyToScrollSupportLegacyUrls: (
+        originalAction: Function,
+        system: Object
+    ): Object => (isShownKey: Object, element: Element): Object => {
+        let scrollToKey = system.layoutSelectors.getScrollToKey();
+
+        if (scrollToKey) {
+            scrollToKey = scrollToKey.toJS();
+
+            if ("operations" === scrollToKey[0] && scrollToKey[1] === isShownKey[1]) {
+                originalAction(scrollToKey, element);
+            }
+        }
+
+        originalAction(isShownKey, element);
```

Comment:
The idea is the following: if someone is directed to one of our pre-swagger urls (e.g. `https://doc.bunq.com/session-server/post_session_server`) there would be no scrolling, because that endpoint does not exist anymore. Instead I decided to let the page scroll to `https://doc.bunq.com/session-server` and then, if the last segment exists, to scroll a bit more to the method. This way we still kinda support legacy urls. I will make it more clear or add a comment then